bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingImpl.java

```diff
@@ -134,6 +135,18 @@ public void setLabel(@Nullable String label) {
         return Collections.unmodifiableList(new ArrayList<>(this.channels));
     }
 
+    @Override
+    public List<Channel> getChannelsOfGroup(String channelGroupId) {
+        List<Channel> channels = new ArrayList<>();
+        for (Channel channel : this.channels) {
+            ChannelUID channelUID = channel.getUID();
+            if (channelUID.isInGroup() && channelUID.getGroupId().equals(channelGroupId)) {
+                channels.add(channel);
+            }
+        }
+        return Collections.unmodifiableList(channels);
```

Comment:
Any reason to use an unmodifiable list as the list is already a newly created one?

Reply:
To be compliant with the `getChannels()` implementation.

https://github.com/eclipse/smarthome/blob/b71e96d5459290a93975c43ae467e01204757e7c/bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingImpl.java#L132-L135

Reply:
It would be happy if we could move forward on #5960.

Okay for me to keep it unmodifiable here just wonder why.

Reply:
> It would be happy if we could move forward on #5960.

Yes, me too.